trafaret/internet.py

```diff
@@ -41,14 +41,29 @@ def email_idna_encode(value):
 
 email_regexp_trafaret = OnError(to_str & Regexp(EMAIL_REGEXP), 'value is not a valid email address')
 email_trafaret = (email_regexp_trafaret | (to_str & email_idna_encode & email_regexp_trafaret))
-Email = to_str & String(allow_blank=True) & OnError(
+Email = to_str & OnError(
     String(max_length=MAX_EMAIL_LEN) & email_trafaret,
     'value is not a valid email address',
     code=codes.IS_NOT_VALID_EMAIL,
 )
 Email = WithRepr(Email, '<Email>')
 
 
+class Hex(RegexString):
```

Comment:
Why not `Hex = WithRepr(to_str & (lambda x: x.lower()) & Regexp(r'^[0-9a-f]*$'), '<Hex>')`?

Reply:
I think if you run that against my test cases, it will fail.
I'm trying to keep the String arguments available.
e.g. `Hex(min_length=2)`
With the composed approach, you've lost the ability to add min_length to the String checker.

Reply:
Yes, I understand and saw test cases. It is just that main trafaret point is composability. If someone needs to check string length, it can be done like this `Hex() & String(max_length=10)`.
I see that first pack of trafarets did not follow this way, but in original form there were not strong monadic composability feature that we have now in form of `&` (and its alias `>>`).

Reply:
I feel like the composability is great to give the user a lot of extensibility.
However, for something already provided by the library, I feel that it's best to provide better efficiency and flexibility. For the starting user, the composability is likely not obvious, so having the string arguments available on specialised forms of strings is useful and helps keep code simple and concise. I'm just not seeing an advantage to composing elements that are in the library itself.